libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/for_loop.hpp

```diff
@@ -0,0 +1,1369 @@
+        friend void tag_invoke(hpx::ranges::for_loop_strided_t, Iter first,
+            Sent last, S stride, Args&&... args)
+        {
+            static_assert(sizeof...(Args) >= 1,
+                "for_loop_strided must be called with at least a function "
+                "object");
+
+            return for_loop_strided(parallel::execution::seq, first, last,
+                stride, std::forward<Args>(args)...);
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename Rng, typename S, typename... Args,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy<ExPolicy>::value &&
+                std::is_integral<S>::value &&
+                hpx::traits::is_range<Rng>::value
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy>::type
+        tag_invoke(hpx::ranges::for_loop_strided_t, ExPolicy&& policy,
+            Rng&& rng, S stride, Args&&... args)
+        {
+            static_assert(sizeof...(Args) >= 1,
+                "for_loop_strided must be called with at least a function "
+                "object");
+
+            using hpx::util::make_index_pack;
+            return parallel::v2::detail::for_loop(
+                std::forward<ExPolicy>(policy), hpx::util::begin(rng),
+                hpx::util::end(rng), stride,
+                typename make_index_pack<sizeof...(Args) - 1>::type(),
+                std::forward<Args>(args)...);
+        }
+
+        // clang-format off
+        template <typename Rng, typename S, typename... Args,
+            HPX_CONCEPT_REQUIRES_(
+                std::is_integral<S>::value &&
+                hpx::traits::is_range<Rng>::value
+            )>
+        // clang-format on
+        friend void tag_invoke(hpx::ranges::for_loop_strided_t, Rng&& rng,
+            S stride, Args&&... args)
+        {
+            static_assert(sizeof...(Args) >= 1,
+                "for_loop_strided must be called with at least a function "
+                "object");
+
+            return for_loop_strided(parallel::execution::seq,
+                hpx::util::begin(rng), hpx::util::end(rng), stride,
+                std::forward<Args>(args)...);
+        }
+    } for_loop_strided{};
+
+    HPX_INLINE_CONSTEXPR_VARIABLE struct for_loop_n_t final
+      : hpx::functional::tag<for_loop_n_t>
+    {
+    private:
+        // clang-format off
+        template <typename ExPolicy, typename Rng, typename Size,
+            typename... Args,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy<ExPolicy>::value &&
+                std::is_integral<Size>::value &&
+                hpx::traits::is_range<Rng>::value
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy>::type
+        tag_invoke(hpx::ranges::for_loop_n_t, ExPolicy&& policy, Rng&& rng,
+            Size size, Args&&... args)
```

Comment:
Wouldn't this be better represented by just taking a range (iter/sent)? Should we add such an overload in the first place?

Reply:
There is an overload taking an explicit iterator and sentinel pair. This is overload would just be convenience for `for_loop(std::begin(c), std::end(c), ...)`.

Reply:
Right. For this reason I'm advocating to not add this particular overload. There is no precedence (IIRC) where a range based algorithm uses the `begin(r)` of the given range `r` only. For instance, `ranges::for_each_n` has no overload that expects a range as its argument, just one expecting an iterator and a count (http://eel.is/c++draft/alg.foreach#lib:for_each_n__).

Reply:
Sorry, my bad. I missed that this was on the `_n` version. Yes, you're right that `for_each_n` doesn't have an equivalent overload and I'm indifferent about keeping this or not. It's a very minor convenience wrapper anyway. So remove?

Reply:
Yes, I'd be in favor of removing this overload.

Reply:
Done.